### PR TITLE
move create-clusters workflow from test-gh-actions

### PR DESCRIPTION
**Pull request by tommartensen**
Moving the create-clusters workflow from test-gh-actions: https://github.com/stackrox/test-gh-actions/blob/main/.github/workflows/create-clusters.yml.

In contrast to #16, this PR does add the workflow as is and does not attempt to rewrite it into an action. 
Both are reusable by other workflows, example: https://github.com/stackrox/test-gh-actions/pull/114/files .

Example build: https://github.com/stackrox/test-gh-actions/actions/runs/5820630385

### CODE REVIEW

**Comment by vikin91** [comment]
There is a danger of clashing names: https://github.com/stackrox/actions/tree/main/infra/create-cluster

Are you planning to use #16 for this later?

**Comment by tommartensen** [comment]
> Are you planning to use https://github.com/stackrox/actions/pull/16 for this later?

No, rewriting the workflow into an action that references other actions through scripts should not be desirable because:
- actions should be small, this workflow is quite complex
- some parts of it (like Slack notifications) would need to be rewritten from using an action to bash

File: .github/workflows/create-cluster.yml [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    if: >- # Skip if no clusters are going to be created.
+      inputs.create-k8s-cluster != 'false' ||
+      inputs.create-os4-cluster != 'false' ||
+      inputs.create-long-cluster != 'false'
+    strategy:
+      matrix:
+        image: [main, scanner, scanner-db, collector]
+    steps:
+      - name: Wait for the ${{matrix.image}} image
+        uses: stackrox/actions/release/wait-for-image@v1
```
Comment (tommartensen) [inline .github/workflows/create-cluster.yml:57]: I am not 100% convinced with hardcoding the reference for the action.
Here is what did not work:
- checking out the repository and using a local action, which would need to pass the workflow ref from the calling workflow to here. For unknown reasons the action behaves differently when called locally or as shell script vs as a remote action, and the `gh_log` output becomes part of the block output [here](https://github.com/stackrox/actions/blob/main/release/wait-for-image/wait-for-image.sh#L33), which causes jq and the loop to fail.
- using an input (or other) variable in the path, e.g. `stackrox/actions/release/wait-for-image@${{ inputs.workflow-ref }}`

The other actions in this workflow are not affected by the output bug. 
If we remove the `gh_log` output from the block `wait-for-image`, we could use the checkout and local action approach (with passing the workflow ref from the calling workflow).

Comment (tommartensen) [reply .github/workflows/create-cluster.yml]: > The other actions in this workflow are not affected by the output bug.
If we remove the gh_log output from the block wait-for-image, we could use the checkout and local action approach (with passing the workflow ref from the calling workflow).

fixed in #18

Comment (tommartensen) [reply .github/workflows/create-cluster.yml]: @vikin91 and I agreed offline to pass the `workflow-ref` to checkout the `actions` repo and run a local action.

**Review by vikin91 — APPROVED** [review]
Only a coarse review, but the moving looks good! Now we have one place to add improvements (I can spot some of them already).

FYI: @JoukoVirtanen

**Pull request merged**